packages/presets/client/src/index.ts

```diff
@@ -69,6 +69,17 @@ export const preset: Types.OutputPreset<ClientPresetConfig> = {
   buildGeneratesSection: options => {
     const reexports: Array<string> = [];
 
+    // the `client` preset is restricting the config options inherited from `typescript`, `typescript-operations` and others.
+    const forwardedConfig = {
+      scalars: options.config.scalars,
+      defaultScalarType: options.config.defaultScalarType,
+      strictScalars: options.config.strictScalars,
+      namingConvention: options.config.scalars,
```

Comment:
It should be mapped to `options.config.namingConvention`

Reply:
Good catch, fixing: https://github.com/dotansimha/graphql-code-generator/pull/8457